portfolio/src/main/webapp/script.js

```diff
@@ -187,3 +187,22 @@ function loadPage(pageNumber) {
   }
   page = pageNumber;  
 }
+
+function checkLogin() {
+  fetch('/login').then(response => response.json()).then(status => {
+    const postAndDeleteElement = document.getElementById('post-and-delete');
+    if(status[0] === 'false'){
+      postAndDeleteElement.innerHTML = '';
```

Comment:
Instead of conditionally building the elements here, please write them out in HTML with `display: none;` set in their style and then change to `display: inline;` inside this if.